index.js

```diff
@@ -178,21 +178,33 @@ var clean = function() {
   return results;
 };
 
-CleanWebpackPlugin.prototype.apply = function(compiler) {
+CleanWebpackPlugin.prototype.apply = function (compiler) {
   var _this = this;
   if (compiler === undefined) {
     return clean.call(_this);
   } else {
+    const pluginName = "clean-webpack-plugin";
     if (_this.options.watch) {
-      compiler.plugin("compile", function(params) {
+      const compile = function (params) {
         clean.call(_this);
-      });
+      }
+      if (compiler.hooks) {
+        compiler.hooks.compile.tap(pluginName, compile);
+      } else {
+        compiler.plugin("compile", compile);
```

Comment:
I would just use `hooks`, if people want to stay on `<4` then do not upgrade the package.

Reply:
Cool. Will do those changes

Reply:
Can I use lambda and other ES6 stuff?

Reply:
@aulisius dont forget Node 6, so as long as you support Node 6.

Also, there is a bunch of changes that are not part of the fix, could you remove those changes so the commit is cleaner.